### ISSUE TITLE
number of rendered images is off in subsequent renders

### ISSUE THREAD

**Issue by xanathon** [n00mkrad/text2image-gui#12]
The problem I reported yesterday on Discord with the beta version of 1.4.0 is the same with the finished version from itch.io: I render a first sequence of three images and get three images, a subsequent render with the same seed and with or without changed parameters only renders two images out of three. See attached image, left top you can see that three images are set, bottom right you can see that only 2/2 were rendered.

In addition to that: the "render" button stays at "cancel" and the progress rotator thingie on the top keeps rotating. Did not happen with the earlier version (1.3.x) 

![image](https://user-images.githubusercontent.com/24546760/191578391-c769cb6e-f86c-4161-89d6-f99c80b7c9de.png)

**Comment by n00mkrad** [comment]
Can you reliably reproduce this?

**Comment by xanathon** [comment]
yes. after every restart and with different phrases and settings I get the same problem.

**Comment by xanathon** [comment]
I have to correct that. If I close the UI and restart, my CUDA device is no longer detected.

**Comment by xanathon** [comment]
That must be a simple for-to or loop-until problem somewhere, because if I increase the number of images to render to four, only three are rendered, if I change it to 5 only 4 are rendered. So maybe somewhere there is a < instead of <= in a for loop or something like that.

**Comment by n00mkrad** [comment]
> That must be a simple for-to or loop-until problem somewhere, because if I increase the number of images to render to four, only three are rendered, if I change it to 5 only 4 are rendered. So maybe somewhere there is a < instead of <= in a for loop or something like that.

That can't be the problem because it works fine for me.

**Comment by xanathon** [comment]
"It works for me" is always a very bad support answer. That may be so but does not invalidate the fact that it does not work here.

Instead of saying "it works for me" (which is never productive and never yields results) you probably should give me additional options to solve this problem. Are there logs somewhere? I do not think something on my system is the problem, since this bug did not happen in 1.3.x, so it must be something that was invented in 1.4.0.
